packages/tools/devtools/devtools-core/src/data-visualization/SharedTreeVisualizer.ts

```diff
@@ -279,10 +344,17 @@ export async function visualizeSharedTreeNodeBySchema(
 				throw new TypeError("Invalid array");
 			}
 
+			const arrayNodeSchemaProperties: Record<string, FieldSchemaProperties> = {};
 			for (const [i, child] of children.entries()) {
-				fields[i] = await visualizeSharedTreeNodeBySchema(
+				arrayNodeSchemaProperties[i] = {
+					allowedTypes: schema.allowedTypes,
+					isRequired: undefined,
+				};
+
+				fields[i] = await visualizeSharedTreeBySchema(
 					child,
-					treeSchema,
+					treeDefinitions,
+					{ allowedTypes: arrayNodeSchemaProperties[i]?.allowedTypes, isRequired: undefined },
```

Comment:
A parking lot question for tomorrow: I see a compile error if I do not include the optional chaining here. Aren't we guaranteed to have `arrayNodeSchemaProperties[i]` because we are assigning one in line `349`?

![image](https://github.com/user-attachments/assets/6189759c-a8cb-4ecd-ab49-fbde863fd369)

Reply:
I think technically the answer is yes in practice, but TS cannot guarantee that something in between didn't cause it to (potentially) disappear from that index. My recommendation so we don't have to use workarounds is to assign that object to a local variable, assign that variable to `arrayNodeSchemaPoperties[i]`, and use it directly in line 357 instead of using an indexed access on the array. Or use `schema.allowedTypes` directly.